lib/srv/db/watcher.go

```diff
@@ -297,14 +299,18 @@ func (c *cloudCredentialsChecker) checkAzure(ctx context.Context, database types
 		return
 	}
 
-	if rid, err := arm.ParseResourceID(database.GetAzure().ResourceID); err == nil {
-		if !slices.Contains(allSubIDs, rid.SubscriptionID) {
-			c.warn(nil, database, fmt.Sprintf("The discovered database %q is in a different subscription (ID: %s) than the database agent's tenant.",
-				database.GetName(),
-				rid.SubscriptionID,
-			))
-			return
-		}
+	rid, err := arm.ParseResourceID(database.GetAzure().ResourceID)
+	if err != nil {
+		c.log.Warnf("Failed to parse resource ID of database %q: %v.", database.GetName(), err)
+		return
+	}
+
+	if !slices.Contains(allSubIDs, rid.SubscriptionID) {
+		c.warn(nil, database, fmt.Sprintf("The discovered database %q is in a different subscription (ID: %s) than the database agent's tenant.",
```

Comment:
last minute nit:
```suggestion
		c.warn(nil, database, fmt.Sprintf("The discovered database %q is in a subscription (ID: %s) that the database agent does not have access to.",
```
subscriptions are in tenants, not the other way around